app/controllers/posts_controller.rb

```diff
@@ -5,6 +5,15 @@ class PostsController < ApplicationController
   # GET /posts.json
   def index
     @posts = Post.all
+    @posts_done_homework = Post.where(done_homework: true, user_id: current_user) 
```

Comment:
Try to use ActiveRecord relations...
For example, if you have the User model linked to a Post, you can use something like this:
```
has_many :posts
```

And in your Post model
```
belongs_to :user
```